sktime_dl/deeplearning/base/estimators/_regressor.py

```diff
@@ -0,0 +1,88 @@
+# Base class for regressors
+
+__author__ = "Withington"
+
+import numpy as np
+import pandas as pd
+
+from sklearn.base import RegressorMixin
+from sklearn.utils.validation import check_is_fitted
+
+from sktime.regressors.base import BaseRegressor
+from sktime.utils.validation.supervised import validate_X, validate_X_y
+
+from sktime_dl.utils import save_trained_model
+
+
+class BaseDeepRegressor(BaseRegressor, RegressorMixin):
+    model_save_directory = None
+    model = None
+    model_name = None
+
+    def build_model(self, input_shape, **kwargs):
+        """
+        Construct a compiled, un-trained, keras model that is ready for training
+        ----------
+        input_shape : tuple
+            The shape of the data fed into the input layer
+        Returns
+        -------
+        output : a compiled Keras Model
+        """
+        raise NotImplementedError('this is an abstract method')
+
+    def predict(self, X, input_checks=True, **kwargs):
+        """
+        Find regression estimate for all cases in X.
+        Parameters
+        ----------
+        X : array-like or sparse matrix of shape = [n_instances, n_columns]
+            The training input samples.
+            If a Pandas data frame of Series objects is passed (sktime format), 
+            a check is performed that it only has one column.
+            If not, an exception is thrown, since this regressor does not yet have
+            multivariate capability.
+        input_checks: boolean
+            whether to check the X parameter
+        Returns
+        -------
+        predictions : 1d numpy array
+            array of predictions of each instance
+        """
+        check_is_fitted(self)
```

Comment:
Perhaps we should make the check_is_fitted test more explicit, checking only for `is_fitted_` attribute? I know that sklearn doesn't have a consistent way of representing the is-fitted state across all estimators and instead follow the trailing underscore convention, but I think I'd prefer to make it more consistent. Any thoughts? We could even make the `check_is_fitted` function a method of some `BaseNetwork`.

Reply:
It seems OK as it is, since ```check_is_fitted``` does check that  ```is_fitted_==True```. I'm not sure that making a more explicit way of doing this would add much?

A network is lacking a final layer, so it can't be fitted. Therefore, I wouldn't add  ```check_is_fitted``` to ```BaseDeepNetwork``` (or a ```BaseNetwork```). If needed, it could be added it ```BaseDeepRegressor``` and to ```BaseDeepClassifier```.

Reply:
Didn't realise sklearn check just looks for trailing _ attributes. That's... funky 

Arguments for explicitly checking is_fitted_ are basically whether we think any trailing _ atts may be made in any methods other than __init__ and fit. They shouldn't and currently don't, but potential

It has a couple other useful messages in the sklearn func and obviously works for that wider toolkit, I reckon just stick with what's here